docs/tutorial/write-a-machine-charm-for-tinyproxy.md

```diff
@@ -0,0 +1,956 @@
+
+### Deploy your charm
+
+As you deploy your charm to Juju, it will be helpful to watch Juju status in real time.
+
+Open another terminal, then run:
+
+```text
+multipass shell juju-sandbox
+```
+
+Next, in the same terminal, run:
+
+```text
+juju status --watch 2s
+```
+
+You should now have two terminals:
+
+- A terminal with working directory `~/tinyproxy`, from earlier.
+
+- A terminal that shows Juju status in real time:
+
+    ```text
+    Model    Controller     Cloud/Region         Version  SLA          Timestamp
+    testing  concierge-lxd  localhost/localhost  3.6.8    unsupported  09:00:00+08:00
+    ```
+
+You're now ready to deploy your charm.
+
+In the `~/tinyproxy` directory, run `juju deploy ./<charm-file>`, where `<charm-file`> is the name of the file created by `charmcraft pack`. For example:
+
+```text
+juju deploy ./tinyproxy_amd64.charm
+```
+
+Juju creates an "application" from your charm. For each unit in the application, Juju starts a machine and installs your charm on the machine. We didn't tell Juju how many units we want, so Juju assumes one unit and starts one machine. After Juju has installed your charm on the machine, Juju starts sending events to your charm so that your charm can install and start tinyproxy.
+
+When your charm has started tinyproxy, the application will go into "active" status:
+
+```text
+Model    Controller     Cloud/Region         Version  SLA          Timestamp
+testing  concierge-lxd  localhost/localhost  3.6.8    unsupported  09:01:38+08:00
+
+App        Version  Status  Scale  Charm      Channel  Rev  Exposed  Message
+tinyproxy  1.11.0   active      1  tinyproxy             0  no
+
+Unit          Workload  Agent  Machine  Public address  Ports  Message
+tinyproxy/0*  active    idle   0        10.71.67.208
+
+Machine  State    Address       Inst id        Base          AZ  Message
+0        started  10.71.67.208  juju-8e7bd9-0  ubuntu@22.04      Running
+```
+
+```{tip}
+For the rest of the tutorial, we'll assume that you're still watching Juju status. To stop watching, press <kbd>Ctrl</kbd> + <kbd>C</kbd>.
+```
+
+### Try the reverse proxy
+
+Now that tinyproxy is running, we can check that it proxies [example.com](http://example.com) on the machine's network.
+
+In your virtual machine, run:
+
+```text
+curl <address>:8000/example/
+```
+
+Where `<address>` is the IP address of machine 0 from Juju status. In our example of Juju status, the IP address is 10.71.67.208.
+
+The output of curl should be similar to:
+
+```text
+<!doctype html><html lang="en"><head><title>Example Domain</title>...
+```
+
+Then run `curl http://example.com` and check that you get the same output.
+
+### Change the configuration
+
+Let's see what happens to the reverse proxy if we change the `slug` configuration option. Run:
+
+```text
+juju config tinyproxy slug=foo
+```
+
+Where `<address>` is the IP address of machine 0 from Juju status.
```

Comment:
This doesn't seem to make sense here (just after the `juju config` command). Probably don't need it again after the above.